### PR TITLE
Updated dpkg to 1.21.7

### PR DESCRIPTION
**Pull request by tjhexf**
Had a lot of trial and error until this completed a build, however, it successfully built after my final commit.
switched the temporary src to the regular again, from debian source
had to remove a patchfile for the script it patched does not exist anymore in 1.12.7
should make upcoming updates a simpler job of just bumping up the version in the build.sh

### CODE REVIEW

File: packages/dpkg/src-statcmd.c.patch [hunk 1]
```diff
@@ -1,14 +0,0 @@
-diff -u -r ../dpkg-1.18.2/src/statcmd.c ./src/statcmd.c
---- ../dpkg-1.18.2/src/statcmd.c	2015-07-12 22:38:47.000000000 -0400
-+++ ./src/statcmd.c	2015-08-25 18:07:59.388890175 -0400
-@@ -161,8 +161,10 @@
- static void
- statdb_node_apply(const char *filename, struct file_stat *filestat)
- {
-+#ifndef __ANDROID__
- 	if (chown(filename, filestat->uid, filestat->gid) < 0)
- 		ohshite(_("error setting ownership of '%.255s'"), filename);
```
Comment (Grimler91) [inline packages/dpkg/src-statcmd.c.patch:10]: Looks like code has been moved, so patch is probably still needed: https://salsa.debian.org/dpkg-team/dpkg/-/blob/main/src/statoverride/main.c#L191

File: packages/dpkg/build.sh [hunk 2]
```diff
@@ -2,11 +2,11 @@ TERMUX_PKG_HOMEPAGE=https://packages.debian.org/dpkg
...
 TERMUX_PKG_LICENSE="GPL-2.0"
 TERMUX_PKG_MAINTAINER="@termux"
-TERMUX_PKG_VERSION=1.21.1
+TERMUX_PKG_VERSION=1.21.7
 TERMUX_PKG_REVISION=1
-#TERMUX_PKG_SRCURL=https://mirrors.kernel.org/debian/pool/main/d/dpkg/dpkg_${TERMUX_PKG_VERSION}.tar.xz
-TERMUX_PKG_SRCURL=https://ftp-osl.osuosl.org/pub/gentoo/distfiles/dpkg_${TERMUX_PKG_VERSION}.tar.xz
-TERMUX_PKG_SHA256=1eb9fd5228b3199284ea5134904bb45b7a5bc12fb044b8e4964d89d2e5bbb563
+TERMUX_PKG_SRCURL=https://mirrors.kernel.org/debian/pool/main/d/dpkg/dpkg_${TERMUX_PKG_VERSION}.tar.xz
+#TERMUX_PKG_SRCURL=https://ftp-osl.osuosl.org/pub/gentoo/distfiles/dpkg_${TERMUX_PKG_VERSION}.tar.xz
```
Comment (Grimler91) [inline packages/dpkg/build.sh:8]: No need to keep old url commented, it can be removed

**Review by Grimler91 — CHANGES_REQUESTED** [review]
Thanks for the PR!

Please squash the commits, and rebase against master branch to get rid of the merge commit.
See for example [this link](https://www.digitalocean.com/community/tutorials/how-to-rebase-and-update-a-pull-request) for instructions on how to rebase.

Preferably the commit message should be something like: `dpkg: update to 1.21.7`

**Comment by tjhexf** [comment]
Thank you! I'll get to these changes right now.

**Pull request closed without merging**